### PR TITLE
Add tests to compare the MATLAB and Python version of the ICLabel network

### PR DESCRIPTION
**Pull request by mscheltienne**
The goal of this PR is to add the tests to compare both networks by themselves.

- [x] Compare weights layer by layer
- [x] Compare network outputs for a fix common input (by-passing ICA decomposition and feature extraction)

### CODE REVIEW

**Comment by adam2392** [comment]
#11 cleaned up the repository considerably. Please merge changes.

**Comment by mscheltienne** [comment]
So one good news, at least as far as the network is concerned, everything looks correct.
If you feed the same input to the forward pass of the network in MATLAB or in PYTHON, you get the same output. 

Any difference lies in the feature extraction or in the feature formatting.

PR ready for merge on my side.

**Comment by mscheltienne** [comment]
@anandsaini024 Can you can provide the MATLAB script you used to generate the input/output in this thread?

**Comment by adam2392** [comment]
> So one good news, at least as far as the network is concerned, everything looks correct. If you feed the same input to the forward pass of the network in MATLAB or in PYTHON, you get the same output.
> 
> Any difference lies in the feature extraction or in the feature formatting.
> 
> PR ready for merge on my side.

Awesome! will take a look later.

File: mne_icalabel/tests/test_iclabel_net.py [hunk 1]
```diff
@@ -1,29 +1,115 @@
...
 
 # load in test data for features from original Matlab ICLabel
 ica_file_path = str(files("mne_icalabel.tests").joinpath("data/eeglab_ica.set"))
 ica_raw_file_path = str(files("mne_icalabel.tests").joinpath("data/eeglab_ica_raw.mat"))
+torch_iclabel_path = Path(__file__).parent.parent / 'assets' / 'iclabelNet.pt'
+matconvnet_iclabel_path = Path(__file__).parent / 'data' / 'netICL.mat'
+mat_image_features_path = Path(__file__).parent / 'data' / 'matlab_images.mat'
+mat_psds_features_path = Path(__file__).parent / 'data' / 'matlab_psds.mat'
+mat_autocorr_features_path = Path(__file__).parent / 'data' / 'matlab_autocorrs.mat'
+mat_labels_file_path = Path(__file__).parent / 'data' / 'matlab_labels.mat'
```
Comment (adam2392) [inline mne_icalabel/tests/test_iclabel_net.py:26]: If you use the `files` as in lines 19-20, I believe it is more robust.

**Review by adam2392 — APPROVED** [review]
One minor comment to robust-ify the data loading, but otw LGTM!

Also, you can run `make black` to run black style formatting on the entire repo to fix style.

**Comment by mscheltienne** [comment]
Oh well, @jacobf18 merged it before I changed the file loading.
Anyway, at some point, it should be swapped for a testing dataset and a fetcher. 

Good to know for `make black`, never heard of it, I usually run a `flake8` workflow. I'll look into it!

**Comment by mscheltienne** [comment]
For now, I've asked Anand to look into feature generation and formatting, let's dig in until we are confident there is no difference between both pipelines 😄

**Comment by adam2392** [comment]
Sounds good!

FYI: https://github.com/jacobf18/iclabel-python/pull/13

@mscheltienne

**Pull request merged**